tgcrypto/aes256.h

```diff
@@ -32,7 +32,7 @@
 #define LROTR(x) (((x) >> 8) | ((x) << 24))
 #define SWAP(x) ((LROTL((x)) & 0x00ff00ff) | (LROTR((x)) & 0xff00ff00))
 #define GET(p) SWAP(*((uint32_t *)(p)))
-#define PUT(ct, st) {*((uint32_t *)(ct)) = SWAP((st));}
+#define PUT(ct, st) {*((uint32_t *)(ct)) = SWAP((st));} (void)0
```

Comment:
Why this?

Reply:
Oh, I put it there to silence empty statement warnings since putting a semicolon after a block creates one. Probably redundant, but one might compile it with werror which will fail due to the empty statement warning, (void)0 makes it explicit

Reply:
I guess we can get rid of semicolons after each `PUT()` usage instead. That should remove the warning and the need of `(void)0` as well.

Reply:
Actually we could simply cast PUT return value to (void) to avoid placing a new block entirely and prevent accidental assignment (I think that's why it's there?)

Reply:
`#define PUT(ct, st) (void)(*((uint32_t *)(ct)) = SWAP((st)))`
This would be the correct use

Reply:
PUT doesn't return anything, or rather, the assignment implicit return value is not being used, the block probably existed to avoid some undesired effects about this behaviour in case PUT was used inside expressions, yet there's no such usage anywhere.
The cast becomes useless at this point and we can just change it to `#define PUT(ct, st) (*((uint32_t *)(ct)) = SWAP((st)))`.

Reply:
Should it be in the header at all then? It's only used in the aes256.c source file, where we can ensure it's used correctly

Reply:
The library works just fine for its purpose, It's not that things are going to change considerably at this point (unless I decide to add support for hardware acceleration), but sure, those macros can live inside aes256.c. The other macros above GET and PUT are their dependencies so they must stay together.